source/core/NumericString.cpp

```diff
@@ -559,33 +582,49 @@ void Format(SubString *format, Int32 count, StaticTypeAndData arguments[], Strin
                             if (precision < 0) {
                                 precision = 0;
                                 truncateSignificant = exponent + 1 - fOptions.Significant;
-                                // need truncate the integer part of the float because the sprintf doesnt do this for us.
+                                // need truncate the inxteger part of the float because the sprintf doesnt do this for us.
                             }
                         }
-                        char asciiReplacementString[2*kTempCStringLength];
-                        Int32 sizeOfNumericString = -1;
                         char formatCode[10];
+                        skipPrev = Int32(strlen(replacementString));
                         if (precision >= 0) {
                             sprintf(formatCode, "%%.*%sf", fOptions.NumericLength);
                             // formatCode : %.*hf
-                            sizeOfNumericString = snprintf(asciiReplacementString, sizeof(asciiReplacementString), formatCode, precision,tempDouble);
+                            sizeOfNumericString = snprintf(replacementString+skipPrev, sizeof(replacementString)-skipPrev, formatCode, precision,tempDouble);
                         } else {
                             sprintf(formatCode, "%%%sf", fOptions.NumericLength);
                             // formatCode: %hf
-                            sizeOfNumericString = snprintf(asciiReplacementString, sizeof(asciiReplacementString), formatCode, tempDouble);
+                            sizeOfNumericString = snprintf(replacementString+skipPrev, sizeof(replacementString)-skipPrev, formatCode, tempDouble);
                         }
-                        Int32 intDigits = (exponent >= 0)? (exponent): 0 ;
-                        RefactorLabviewNumeric(&fOptions, asciiReplacementString, &sizeOfNumericString, intDigits, truncateSignificant);
-                        UpdateNumericStringWithDecimalSeparator(fOptions, asciiReplacementString, sizeOfNumericString);
-                        buffer->Append(sizeOfNumericString, (Utf8Char*)asciiReplacementString);
+                        sizeOfNumericString += skipPrev;
+                        if (exponent > 0)
+                            intDigits = exponent;
+                        // further processing done by shared code which calls RefactorLVNumeric below switch
                         argumentIndex++;
                     }
                     break;
                     case 'e': case 'E':
                     {
                         TypeRef argType = arguments[argumentIndex]._paramType;
-                        Double tempDouble = ReadDoubleFromMemory(argType,  arguments[argumentIndex]._pData);
+                        void *pData = arguments[argumentIndex]._pData;
```

Comment:
I had already tried to do this but didn't like what I came up with because it accesses 5 different local variables which have to be passed, plus has a continue which needs to be converted to a return value.  ArgumentIndex and parseFinished really are specific to the context and don't make much sense when moved into a helper function, so only the argType and reload and pData really make sense to be factored out, and then it doesn't seem worth it.